### ISSUE TITLE
[extras] test section vs test/Project.toml

### ISSUE THREAD

**Issue by oxinabox** [JuliaBesties/BestieTemplate.jl#408]
### Description

There are two ways to specify tests dependencies
`test/Project.toml`  and the `[extras]` section with `targets=test`.

I prefer the later as it makes it impossible to get incompatible versions between tests and main dependencies,
and harder to have duplicates.
Further technically the `test/Project.toml` is kinda experimental (see the [note here](https://pkgdocs.julialang.org/v1/creating-packages/#Alternative-approach:-test/Project.toml-file-test-specific-dependencies)) _"The exact interaction ... are not fully worked out and may be subject to change in future versions"_
and the original developer of that feature is no longer contributing.

Currently BestieTemplates.jl does the former.
I believe PkgTemplates.jl lets you chose, and by default does the later.

**Comment by abelsiqueira** [comment]
Yeah, I was under the impression that `test/Project.toml` would become the norm, but it does seem to have stalled. A developer leaving explains a lot.

It makes sense to me to have a separate environment for the `test` like the `docs`, but indeed, it lacks tooling to have it working nicely. I will have to look into this with more detail to have an opinion on changing the default.

Having it as an option is surely possible.

**Comment by mmikhasenko** [comment]
I always used `[targets] test=`, and discovered the way of `test/Projects.toml` with the `BestieTemplate`.

What I noticed is that `test/Projects.toml` takes the preference by `julia`: in case `test/Projects.toml` exists, `[targets] test=` is ignored.

**Comment by abelsiqueira** [comment]
The `test/Project.toml` approach has been marked as the recommended approach in
https://pkgdocs.julialang.org/v1/creating-packages/#Test-specific-dependencies so I'm closing this issue.

If an option supporting this is still required let me know. An option in the testing strategy like `legacy` could be implemented by someone.
